Assets/Scripts/StarComponent.cs

```diff
@@ -60,23 +76,46 @@ public void SetStarScale(float magnitudeScale)
     {
         var magScaleValue = DataManager.Instance.GetRelativeMagnitude(starData.Mag) * magnitudeScale;// ((starData.Mag * -1) + maxMagnitude + 1) * magnitudeScale;
         sceneInitialScale = initialScale * magScaleValue;
-        transform.localScale = sceneInitialScale;
+        StarVisible.transform.localScale = sceneInitialScale;
+    }
+
+    // this is called when another star is selected
+    private void StarSelected(Star selectedStarData, string playerName, Color playerColor)
+    {
+        if (selectedStarData.Hipparcos != starData.Hipparcos && FloatingInfoPanel)
+        {
+            if (FloatingInfoPanel.GetComponent<FloatingInfoPanel>().playerName == playerName)
+                Destroy(FloatingInfoPanel);
+        }
+    }
+    private void SunSelected(bool selected)
+    {
+        if (FloatingInfoPanel)
+            Destroy(FloatingInfoPanel);
+    }
+    private void MoonSelected(bool selected)
+    {
+        if (FloatingInfoPanel)
+        {
+            // FloatingInfoPanel.GetComponent<FloatingInfoPanel>().playerName !=

```

Comment:
Looks like you forgot to delete this commented-out code.

Reply:
Thanks for catching this!